src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -0,0 +1,179 @@
+			if (this.focusedCard && !this.selectedCard) {
+				this.selectCard(this.focusedCard);
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+		else if (e.keyCode === KeyCode.LeftArrow || e.keyCode === KeyCode.UpArrow) {
+			if (this.focusedCard) {
+				this.selectCard(this.findPreviousCard(this.focusedCard));
+			}
+			DOM.EventHelper.stop(e, true);
+		} else if (e.keyCode === KeyCode.RightArrow || e.keyCode === KeyCode.DownArrow) {
+			if (this.focusedCard) {
+				this.selectCard(this.findNextCard(this.focusedCard));
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+	}
+
+	private findPreviousCard(currentCard: azdata.RadioCard): azdata.RadioCard {
+		const currentIndex = this.cards.indexOf(currentCard);
+		const previousCardIndex = currentIndex === 0 ? this.cards.length - 1 : currentIndex - 1;
+		return this.cards[previousCardIndex];
+	}
+
+	private findNextCard(currentCard: azdata.RadioCard): azdata.RadioCard {
+		const currentIndex = this.cards.indexOf(currentCard);
+		const nextCardIndex = currentIndex === this.cards.length - 1 ? 0 : currentIndex + 1;
+		return this.cards[nextCardIndex];
+	}
+
+	public get cards(): azdata.RadioCard[] {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, azdata.RadioCard[]>((props) => props.cards, []);
+	}
+
+	public get cardWidth(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardWidth, undefined);
+	}
+
+	public get cardHeight(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.cardHeight, undefined);
+	}
+
+	public get iconWidth(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconWidth, undefined);
+	}
+
+	public get iconHeight(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.iconHeight, undefined);
+	}
+
+	public get selectedCardId(): string | undefined {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string | undefined>((props) => props.selectedCardId, undefined);
+	}
+
+	public set selectedCardId(newValue: string | undefined) {
+		this.setPropertyFromUI<azdata.RadioCardGroupComponentProperties, string | undefined>((props, value) => props.selectedCardId = value, newValue);
+	}
+
+	public getIconClass(card: azdata.RadioCard): string {
+		if (!this.iconClasses[card.id]) {
+			this.iconClasses[card.id] = `cardIcon icon ${getIconClass(card.icon)}`;
+		}
+		return this.iconClasses[card.id];
+	}
+
+	public setProperties(properties: { [key: string]: any }) {
+		super.setProperties(properties);
+		if (this.selectedCardId) {
+			const filteredCards = this.cards.filter(c => { return c.id === this.selectedCardId; });
+			if (filteredCards.length === 1) {
+				this.selectCard(filteredCards[0]);
```

Comment:
Why this extra logic? It would make sense to me that you'd just select the first one if there were duplicates - in that case just using find seems appropriate.

Reply:
adding handler in case there are no or multiple cards matching the giving ID.